presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java

```diff
@@ -195,8 +194,16 @@ private void logTaskInfos(List<SqlTask> sqlTasks)
                     Long.toString(taskIOStats.getOutputPositions().getTotalCount())));
         }
 
-        for (String row : StringTableUtils.getTableStrings(taskLogInfoList)) {
-            log.info(row);
+        logInfoTable(taskLogInfoList);
+    }
+
+    private void logTaskInfosByQueryID(Map<QueryId, List<SqlTask>> queryIDToSqlTaskMap)
+    {
+        for (Map.Entry<QueryId, List<SqlTask>> mapEntry : queryIDToSqlTaskMap.entrySet()) {
+            // Log the Query
+            logQueryInfo(sqlTaskManager.getQueryContext(mapEntry.getKey()));
```

Comment:
Sorry I think I changed my mind (only a little) on this. Run through this loop twice. First time just log the query info. Second time just log the task infos. Each task info contains the query ID so someone can look at the queries and then look at tasks from individual queries.

The tasks still need to be grouped by query so this isn't wasted work.

There could be O(100) queries. You need to be able to just see just the queries first and then separately do a text search for the query id to find the tasks. That's what the example I gave showed.

Reply:
Also one more pony... can you sort the queries before you log them in descending order by max total memory?